src/openrct2/util/Prefetch.h

```diff
@@ -0,0 +1,34 @@
+/*****************************************************************************
+ * Copyright (c) 2014-2024 OpenRCT2 developers
+ *
+ * For a complete list of all authors, please refer to contributors.md
+ * Interested in contributing? Visit https://github.com/OpenRCT2/OpenRCT2
+ *
+ * OpenRCT2 is licensed under the GNU General Public License version 3.
+ *****************************************************************************/
+
+#if defined(__amd64__) || defined(_M_AMD64) || defined(__i386__) || defined(_M_IX86)
+// Don't bother checking for CPUID, prefetch is available since Pentium 4
+#    include <xmmintrin.h>
+// This cannot be expressed as `constexpr` function, exclude it from clang-tidy check
+// NOLINTNEXTLINE(cppcoreguidelines-macro-usage)
+#    define PREFETCH(x) _mm_prefetch(reinterpret_cast<const char*>(x), _MM_HINT_T0)
+
+#elif defined(_MSC_VER) && defined(_M_ARM64)
+// ARM64 prefetch is available since ARMv8.
+// MSVC's help (https://learn.microsoft.com/en-us/cpp/intrinsics/arm64-intrinsics?view=msvc-170)
+// is useless and points to a different header (arm64_neon.h), which doesn't contain prefetch.
```

Comment:
Please leave (strong) opinions out of the codebase. 
```suggestion
// is incorrect and points to a different header (arm64_neon.h), which doesn't contain prefetch.
```